### DISCUSSION TITLE
About your recent JOSS article

### DISCUSSION THREAD

**Discussion by j3r3m1** [IMMM-SFA/naturf#178] [General]
We are a bit annoyed by your article recently published by JOSS (https://joss.theoj.org/papers/10.21105/joss.06712.pdf). This paper discusses a new tool equivalent to the GeoClimate tool that has been previously described in JOSS as well (https://joss.theoj.org/papers/10.21105/joss.03541).

GeoClimate has been identified as a similar tool (which is fine) but the comparison contains many erroneous information about GeoClimate. Below we have highlighted each of them and described what was wrong. We have warned the editors about this issue but they will not make any modification. If you feel you can do something to change it, wonderful. If not, at least we think that it is interesting that you better know about GeoClimate.

> Similar to NATURF, the open-source toolbox GeoClimate (Bocher et al., 2021) aims to quantify the effect of urban features for climate models. Both tools provide outputs for WRF to be used with the NUDAPT/WUDAPT framework, but at different resolutions. GeoClimate is intended for the production of local climate zones, data classified as level 0 by WUDAPT used to approximate urban parameters within the zone (Ching et al., 2018).

--> You say that the local climate zones are calculated and then used to approximate urban parameters within a zone. It is the contrary: we calculate urban parameters that can then be directly used in numerical models and we also calculate Local Climate Zones from these indicators (Figure 1 of the GeoClimate published paper in JOSS cannot be interpreted differently).

> NATURF produces level 2 data, precise urban parameter data at finer resolutions. The input data used by each tool influences the resolution of outputs; GeoClimate primarily uses OpenStreetMap data where building heights must be estimated while NATURF uses shapefiles with building height and footprint data included.

--> GeoClimate can use any dataset containing building height and footprint data using the generic workflow. One just has to convert attributes from its input data to the corresponding generic attribute names. Figure 4 of the article shows the generic workflow which is used with OpenStreetMap and BDTopo data.

> For both tools, study areas at the neighborhood-scale have comparable memory usage and computational time. For larger areas, particularly at the city-scale, NATURF is more suited for use on high-performance computers (HPC) due to the number and complexity of interactions being calculated.

--> We are unsure to interpret well this sentences. Do the authors solely focus on the NATURF saying that due to the number and complexity of interactions being calculated it is more appropriate to run it on HPC ? Or do they mean that NATURF is more suited than GeoClimate for use on HPC ? If the second, there is nothing allowing to state such information. GeoClimate has been run over Europe using HPC several times and has shown good performances ([Wurtz et al., 2024])( https://meetingorganizer.copernicus.org/EGU24/EGU24-8061.html)).

> Despite this, installation and execution of NATURF requires only a few minutes and edits to code, and the use of hamilton allows for the tracking of code execution progress. Additionally, NATURF can internally produce outputs ready for WRF whereas GeoClimate requires the installation and use of additional packages.

--> It seems NATURF needs the installation of Python packages. GeoClimate needs solely to have Java > 11 installed on the computer, no other requirements or installations.

> While NATURF and GeoClimate both produce urban parameter data for urban microclimate modeling, NATURF does so at a higher resolution with flexibility to analyze complex interactions between buildings at multiple scales.

--> The resolution is clearly the same since the intention is the same...

Of course we do not claim for having the best software in the field but we are claiming for truth being written in scientific journal. We are a bit annoyed that such erroneous informations can be published by JOSS. Several of us suscribed as reviewer for JOSS, and even though you (or the editor) did not think having us as a reviewer may have been a good idea, I think it would be good to add contributors of software cited in a JOSS article as reviewer of a given article (but having the specific role of checking that no erroneous information is written about their work…).

The authors of the JOSS article "GeoClimate: a Geospatial processing toolbox for environmental and climate studies".

**Comment by levisweetbreu** [comment]
Hello,

We appreciate your clarification on the procedures and workings of GeoClimate, and we value the contribution of your software to the broader urban modeling community. We did not intend to slight your work in our paper but rather acknowledge the differences between the two tools and the different research applications they address. The work you and others in this community have done were inspirations to create NATURF to address our intended research objectives. As such, we do not feel that any edits to our paper are necessary. These two tools are complementary to each other, and we do not make any claim otherwise.  

Best,

The NATURF team
